projects/core/directives/dropdown/dropdown-context.directive.ts

```diff
@@ -38,7 +38,7 @@ export class TuiDropdownContextDirective extends TuiDriver implements TuiRectAcc
         this.stream$.next(true);
     }
 
-    @HostListener('document:click.silent', ['$event.target'])
```

Comment:
It should remain silent, otherwise it triggers change detection on every click on the document. Instead please add `@shouldCall` decorator and move if condition to it. It's a decorator from `@tinkoff/ng-event-plugins`, it takes a predicate as an argument and if it returns `true`, calls the method insize `ngZone.run`. You can access `this` inside that predicate.

Reply:
cool, thx